### PR TITLE
adding ecotax to product variant

### PR DESCRIPTION
**Pull request by bogdan202**
allow to add the ecotax to a product variant in the module for PrestaShop

### CODE REVIEW

**Review by clotairer — APPROVED** [review]

**Comment by clotairer** [comment]
Linked to this feature : 
https://github.com/shoppingflux/module-prestashop/pull/176

**Comment by budrea** [comment]
The ecotax info is something only available at the product (parent) level. The variation (child) inherits this information. 

If we want to change this behaviour, we need to have a discussion at the Product level (at Shoppingfeed) and see what this implies. So I cannot validate this PR, for now. 

Thanks.

**Review by budrea — APPROVED** [review]
We're in the process of making the ecotax property available at the variation level, so we can now approve this PR. Thank you for the proposal. 

It will be released next week.

File: src/Xml/XmlProductFeedWriter.php [hunk 1]
```diff
@@ -129,6 +129,10 @@ private function writeSharedProduct(Feed\Product\AbstractProduct $product)
         $this->writeElement('reference', $product->getReference());
         $this->writeElement('quantity', $product->getQuantity());
         $this->writeElement('price', $product->getPrice());
+
+        if ($ecotax = $product->getEcotax()) {
```
Comment (william-leny) [inline src/Xml/XmlProductFeedWriter.php:133]: there is an extraneous `ecotax` element in `writeProduct` method.
tested with `php tests/functional/products-random.php feed.xml 1` (set the ecotax to product in the script)

**Review by william-leny — CHANGES_REQUESTED** [review]

**Comment by budrea** [comment]
I spoke too soon, we won't be supporting the ecotax property at the child / variation level just yet. So I'm closing the PR for now. 

Thanks again for the proposal. If we decide to support this, I'll revive it (or propose a new PR, re-using your code).

**Pull request closed without merging**